packages/components/src/2016-12-01/button.js

```diff
@@ -27,13 +28,75 @@ class Button extends HTMLElement {
     shadowRoot.innerHTML = `
       <orion-inline
         border-radius="2"
-        background="black"
-        color="white"
+        background="white"
+        color="black"
         padding-horizontal="3"
         padding-vertical="2"
-        dim
         pointer><slot /></orion-inline>
     `;
+    this.shadowEl = shadowRoot.children[0];
+
+    this.addEventListener('mouseenter', () => {
+      if (this.state.disabled) { return; }
+      const nextState = ButtonState.enterHover(this.state);
+      this.dispatchEvent(new CustomEvent('change', { detail: nextState }));
```

Comment:
![control flow v1](https://cloud.githubusercontent.com/assets/178653/21732668/697ea4b6-d40f-11e6-8b94-54c153c3d232.png)

I meant that dispatch to enable the controlled button component to call-back its controller when its state changes (step 3.). I think of it as internal—not for consumption by the client app.

The mouseenter event that kicked off this flow will still propagate up for the client to consume. (Step 2a.)

I did it this way since attributes can't hold functions. BUT since we want to set properties instead or attributes I can try doing this with a callback set on a property instead of this event-based approach.